qiskit_nature/algorithms/pes_samplers/bopes_sampler.py

```diff
@@ -89,19 +104,41 @@ def __init__(
                 raise QiskitNatureError(
                     "num_bootstrap must be None or an integer greater than or equal to 2"
                 )
-
-        if isinstance(self._gss.solver, VariationalAlgorithm):  # type: ignore
+        if self._is_variational_solver:
             # Save initial point passed to min_eigensolver;
             # this will be used when NOT bootstrapping
-            self._initial_point = self._gss.solver.initial_point  # type: ignore
+            self._initial_point = self._state_solver.solver.initial_point
 
-    def sample(self, problem: BaseProblem, points: List[float]) -> BOPESSamplerResult:
+    def sample(
+        self,
+        problem: BaseProblem,
+        points: List[float],
+        aux_operators: Optional[
+            ListOrDictType[
+                Union[
+                    SecondQuantizedOp,
+                    PauliSumOp,
+                    Callable[[BaseProblem], Union[SecondQuantizedOp, PauliSumOp]],
```

Comment:
I see a unit test that passes in a SecondQuantizedOp, maybe since the conversion is done elsewhere the direct passing in of a qubit operator (PauliSumOp) is already tested as such, though it could not harm to do so from this level. Also the callable path through the code is not exercised by unit test and would be good to cover.

Reply:
the test now checks the three possible types: Callable, SecondQuantizedOp, PauliSumOp